src/Step.js

```diff
@@ -3,32 +3,49 @@ import { useEffect, createContext, useContext } from 'react';
 
 export const StepContext = createContext<Object>(null);
 
-type Props = {
+type Props = {|
   children: React$Node,
-} & Losen$Step;
-
-const Step = ({ children, name, validator, autoSkip }: Props) => {
-  const { registerStep, activeStep, updateStep, initialized } = useContext(
-    StepContext,
-  );
-
-  const stepInfo = {
-    name,
-    validator,
-    autoSkip,
-  };
+  ...Losen$Step,
+|};
+
+const Step = ({ children, name, validator, autoSkip, state }: Props) => {
+  const {
+    registerStep,
+    activeStep,
+    updateStep,
+    initialized,
+    stateManager,
+  } = useContext(StepContext);
 
   useEffect(() => {
     if (!initialized) {
-      registerStep(stepInfo);
+      registerStep({
+        name,
+        validator,
+        autoSkip,
+      });
     }
-  }, [name]);
+  }, [name, validator, autoSkip, initialized, registerStep]);
 
   useEffect(() => {
     if (initialized) {
-      updateStep(stepInfo);
+      updateStep({
+        name,
+        validator,
+        autoSkip,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [autoSkip, validator, initialized]);
+
+  useEffect(() => {
+    if (state && stateManager) {
+      state.forEach(item => {
+        const key = Object.keys(item)[0];
```

Comment:
Consider for readability

```suggestion
        const [key, value] = Object.entries(item)[0];
```

Entries requires a polyfill for old browsers though. It's added in otovoweb.